v0.5/fastn-p1/src/utils.rs

```diff
@@ -81,6 +81,10 @@ impl fastn_p1::Kind {
             return None;
         }
 
+        self.to_identifier_()
+    }
+
+    pub fn to_identifier_(&self) -> Option<fastn_p1::Identifier> {
```

Comment:
Why? Is this performance optimisation or you do not want those checks?

Reply:
`to_identifier_` is used here:
```rs
if let Some(v) = self.to_identifier_() {
    o.insert("name".into(), v.debug(source));
```
This helps in better JSON for testing the kind with arguments.

Reply:
Why can't that use `to_identifier()`?

Reply:
Because it has 'args' and there's a check in `to_identifier` which makes it return `None`

Reply:
That check is to make sure you are not losing data. Which check if failing? Why are you ignoring that bit of data?

Reply:
I think our `qualified_identifier` parser is wrong. 

From: 

```rs
        t!("foo", {"module": {"package": "foo"}});   // <-- wrong
        // t!("foo", {"module": "foo"}); <-- correct
```

Or we need some other term.